internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,535 @@
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.Query(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "query fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: response.Status.Reason})
+	} else {
+		outputData, err := buildQueryResp(int64(0), httpReq.OutputFields, response.FieldsData, nil)
+		if err != nil {
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "show result by row wrong", "originData": response.FieldsData, HTTPReturnError: err.Error()})
+		} else {
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: outputData})
+		}
+	}
+}
+
+func (h *Handlers) get(c *gin.Context) {
+	httpReq := GetReq{
+		DbName: DefaultDbName,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	coll, err := h.describeCollection(c, httpReq.DbName, httpReq.CollectionName, false)
+	if err != nil || coll == nil {
+		return
+	}
+	body, _ := c.Get(gin.BodyBytesKey)
+	filter, err := checkGetPrimaryKey(coll.Schema, gjson.Get(string(body.([]byte)), "id"))
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "make sure the collection's primary field", HTTPReturnError: err.Error()})
+		return
+	}
+	req := milvuspb.QueryRequest{
+		CollectionName:     httpReq.CollectionName,
+		Expr:               filter,
+		OutputFields:       httpReq.OutputFields,
+		GuaranteeTimestamp: BoundedTimestamp,
+	}
+	username, _ := c.Get(ContextUsername)
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.Query(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "query fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: response.Status.Reason})
+	} else {
+		outputData, err := buildQueryResp(int64(0), httpReq.OutputFields, response.FieldsData, nil)
```

Comment:
The second parameter of "buildQueryResp" should not be directly derived from "httpReq.OutputFields"; rather, it should be calculated based on "response".